bears/general/HTTPSBear.py

```diff
@@ -0,0 +1,72 @@
+from urllib.parse import urlparse
+
+from bears.general.URLBear import URLBear
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+
+from coalib.results.Diff import Diff
+from coalib.settings.Setting import typed_dict
+
+
+class HTTPSBear(LocalBear):
+    DEFAULT_TIMEOUT = 15
+    LANGUAGES = {'All'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Documentation'}
+    BEAR_DEPS = {URLBear}
+
+    def run(self, filename, file, dependency_results=dict(),
+            network_timeout: typed_dict(str, int, DEFAULT_TIMEOUT)=dict()):
+        """
+        Find http links in any text file and check if the https version of
+        link is valid. If so, provide option for replacing it with https.
+
+        An https link is considered valid if the server responds with a 2xx
+        code.
+
+        Warning: This bear will make HEAD requests to all URLs mentioned in
+        your codebase, which can potentially be destructive. As an example,
+        this bear would naively just visit the URL from a line that goes like
+        `do_not_ever_open = 'https://api.acme.inc/delete-all-data'` wiping out
+        all your data.
+
+        :param dependency_results:
+            Results given by URLBear.
+        """
+        for result in dependency_results.get(URLBear.name, []):
+            line_number, link, code, context = result.contents
+            if (link[0:5] != 'https'):
+                httpslink = 'https' + link[4:]
+                host = urlparse(httpslink).netloc
+                network_timeout = {urlparse(url).netloc
+                                   if not url == '*' else '*': timeout
+                                   for url, timeout in network_timeout.items()}
+                httpscode = URLBear.get_status_code(
+                    httpslink,
+                    network_timeout.get(host)
+                    if host in network_timeout
+                    else network_timeout.get('*')
+                    if '*' in network_timeout
+                    else URLBear.DEFAULT_TIMEOUT)
+                if httpscode and 200 <= httpscode < 300:
+                    diff = Diff(file)
+                    current_line = file[line_number - 1]
+                    start = current_line.find(link)
```

Comment:
the link position is not passed by `URLBear`? In that case alright, but let's file an issue afterwards to improve `URLBear` so it also gives us those (as it probably does the same procedure here to actually find links).